test/cpp/test_ir.cpp

```diff
@@ -181,6 +182,31 @@ TEST_F(IrTest, TestSizeAddNode) {
   });
 }
 
+TEST_F(IrTest, TestSizeAddNodeDynamic) {
```

Comment:
nit: maybe, 
* add a recursive `(a.size(0) + a.size(1)) + a.size(0)` ?
* use different tensors `a.size(0) + b.size(1)`

I wouldn't push for these tests though

Reply:
Added a test for using different tensors.